### PR TITLE
release - automation

### PR DESCRIPTION
**Pull request by Incpi**
Hello @dbeck121,

I've added the below file to the action workflow and have the same automation for my extension (adding here).

What it does is added in the comments file. There's no need to release manually. Direct data extraction from the manifest file will occur.

To put it briefly, when you update the bin folder in the main branch using a zip file, the release will attach the same zip file and extract the data metadata directly from the file.

No need to add "GitHub token" it will take care by action.
___________

image
Bin folder files are as below
image

#### Delete bin folder content for initial version or later from 1st bot release, then it will work as expected. I am not deleting from git automation (workflow only updates).
__________

zipFiles.sh will directly create a 2 zip files in bin folder (nothing extra / no switch)

Thanks

### CODE REVIEW

**Comment by dbeck121** [comment]
@Incpi great job

I made the script compatible to mac. grep -P does not work here.

Also I think it is nice to create bin files on dev push and release on main push... what do you think?

**Comment by Incpi** [comment]
Hey, 

- I think someone/merge PR made change in dev will generate each time. as long as it's not creating PR to main. we are good.
- If i understood correctly PR will be manual to main.
- Then, we good we can do. So developer can test newly generated file directly before creating PR.

Thanks

**Comment by dbeck121** [comment]
Oh... Left some German comments... Thanks for changing :)

**Review by dbeck121 — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      # Ensures that the zip file is uploaded with the correct name and content type
+      - name: Attach zip files to release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Use the GitHub token for authentication
+        run: |
+          IFS='|' read -r -a ZIP_ARRAY <<< "${{ env.ZIP_FILES }}"  # Split the pipe-separated list into an array
+          for ZIP_FILE in "${ZIP_ARRAY[@]}"; do
+            ZIP_BASENAME=$(basename "$ZIP_FILE")  # Get the clean zip file name
+            echo "Attaching $ZIP_BASENAME"
+            gh release upload "Version_${{ env.CLEAN_VERSION }}" "$ZIP_FILE" --clobber
```
[the remark this answers is not in the document]
Comment (Incpi) [reply .github/workflows/release.yml:97]: Yes we can.

**Comment by Incpi** [comment]
@dbeck121 build.yml need name change like create zip from dev branch

**Pull request merged**